plugins/homepagecanonical/class.homepagecanonical.plugin.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Homepage Canonical Plugin.
+ *
+ * @author Patrick Kelly <patrick.k@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GNU GPLv2 http://www.opensource.org/licenses/gpl-2.0.php
+ */
+
+class HomepageCanonicalPlugin extends Gdn_Plugin {
+
+    /**
+     * If the Default Controller page is requested (Discussions or Categories), 301 redirect to "/" and insert the Canonical tag.
+     *
+     * @param VanillaController $sender
+     * @param array $args
+     */
+    public function base_render_before($sender, $args) {
+        // Use $SERVER['REQUEST_URI'], and not request()->path() because we alter path;
+        $requestURI = Gdn::request()->getValueFrom('server', 'REQUEST_URI');
```

Comment:
`getValueFrom()` is essentially `val()` for the request object. Elsewhere I see us using `Gdn::request()->getRequestArguments('server')['REQUEST_URI']` so I'd roll with that.